File: 5 Kyu/RGB To Hex Conversion/solution.js
function rgb(r, g, b){
    let res = "";
    for (let n of [r,g,b]) {
        const first = n < 0 ? 0 : Math.min(n / 16, 15.9375);
        const second = (first % 1) * 16;
        res += Math.floor(first).toString(16) + second.toString(16);
    }
    return res.toUpperCase();
}

File: 5 Kyu/RGB To Hex Conversion/solution.go
package kata

import (
  	"math"
  	"strings"
  	"strconv"
)

func RGB(r, g, b int) string {
  	return Hex(float64(r)) + Hex(float64(g)) + Hex(float64(b))
}

func Hex(num float64) string {
    first := math.Max(0, num)
    first = math.Min(first/16, 15.9375)
    second := math.Mod(first, 1) * 16

    return strings.ToUpper(strconv.FormatInt(int64(math.Floor(first)), 16) + strconv.FormatInt(int64(second), 16))
}